Apps/UnitTests/X11/App.cpp

```diff
@@ -0,0 +1,64 @@
+#define XK_MISCELLANY
+#define XK_LATIN1
+#include <X11/Xlib.h> // will include X11 which #defines None... Don't mess with order of includes.
+#include <X11/Xutil.h>
+#undef None
+#include "../Shared/Tests.cpp"
+
+static const char* s_applicationName  = "Babylon Native Unit Tests";
+
+int main() {
+    XInitThreads();
+    Display* display = XOpenDisplay(NULL);
+    int32_t screen = DefaultScreen(display);
+    int32_t depth  = DefaultDepth(display, screen);
+    Visual* visual = DefaultVisual(display, screen);
+    Window root   = RootWindow(display, screen);
+    const int width = 640;
+    const int height = 480;
+
+
+    XSetWindowAttributes windowAttrs;
+    windowAttrs.background_pixel = 0;
+    windowAttrs.background_pixmap = 0;
+    windowAttrs.border_pixel = 0;
+    windowAttrs.event_mask = 0
+            | ButtonPressMask
+            | ButtonReleaseMask
+            | ExposureMask
+            | KeyPressMask
+            | KeyReleaseMask
+            | PointerMotionMask
+            | StructureNotifyMask
+            ;
+
+    Window window = XCreateWindow(display
+                            , root
+                            , 0, 0
+                            , width, height, 0
+                            , depth
+                            , InputOutput
+                            , visual
+                            , CWBorderPixel|CWEventMask
+                            , &windowAttrs
+                            );
+
+    // Clear window to black.
+    XSetWindowAttributes attr;
+    memset(&attr, 0, sizeof(attr) );
+    XChangeWindowAttributes(display, window, CWBackPixel, &attr);
+
+    const char* wmDeleteWindowName = "WM_DELETE_WINDOW";
+    Atom wmDeleteWindow;
+    XInternAtoms(display, (char **)&wmDeleteWindowName, 1, False, &wmDeleteWindow);
+    XSetWMProtocols(display, window, &wmDeleteWindow, 1);
+
+    XMapWindow(display, window);
+    XStoreName(display, window, s_applicationName);
```

Comment:
This is probably a copy and paste, but this code is a bit inconsistent. There are spacing issues all over and constants defined sometimes with the code (wmDeleteWindowName) and sometimes outside (s_applicationName).

Reply:
Okay, I cleaned it up a bit. I can do the same for the ValidationTests/Playground if desired